### PR TITLE
Fix restore repo bug, clarify the problem of ForeignIndex

### LINKED ISSUE #22581: restore-repo: issue comments are no longer imported
### Description

Since https://github.com/go-gitea/gitea/commit/71ca3067bcc6c7c7772d38fc7590505c8c7148ed issue comments are no longer imported with `gitea restore-repo`.

See `func (r *RepositoryRestorer) GetComments(...)` in `services/migrations/restore.go` , it still uses `commentable.GetForeignIndex()` ([link](https://github.com/go-gitea/gitea/blob/29b78bce3336fcedc08d17b71a52e593d65cd951/services/migrations/restore.go#L201)) but as far as I can see that foreign index isn't loaded any more since that commit.

This patches things up for me locally, but I don't know if this is the right thing to do:

```diff
diff --git a/services/migrations/restore.go b/services/migrations/restore.go
index fd337b22c..6ee878a64 100644
--- a/services/migrations/restore.go
+++ b/services/migrations/restore.go
@@ -198,7 +198,7 @@ func (r *RepositoryRestorer) GetIssues(page, perPage int) ([]*base.Issue, bool,
 // GetComments returns comments according issueNumber
 func (r *RepositoryRestorer) GetComments(commentable base.Commentable) ([]*base.Comment, bool, error) {
        comments := make([]*base.Comment, 0, 10)
-       p := filepath.Join(r.commentDir(), fmt.Sprintf("%d.yml", commentable.GetForeignIndex()))
+       p := filepath.Join(r.commentDir(), fmt.Sprintf("%d.yml", commentable.GetLocalIndex()))
        _, err := os.Stat(p)
        if err != nil {
                if os.IsNotExist(err) {
```

To reproduce the issue, unzip [demo-repo-22581.zip](https://github.com/go-gitea/gitea/files/10511505/demo-repo-22581.zip), then run `gitea restore-repo -r /path/to/demo-repo-22581 --owner_name youruser --repo_name demo-repo-22581`

When visiting the repo in Gitea's web interface, you'll notice that there is one imported issue, but no comments, even though there is a corresponding file in `demo-repo-22581/comment/104010.yml`.

There's nothing about this in the logs.

### Gitea Version

29b78bce3336fcedc08d17b71a52e593d65cd951

### How are you running Gitea?

Self-built from the `main` branch.

### Database

PostgreSQL

### PR DESCRIPTION
**Pull request by wxiaoguang**
Fix #22581

TLDR: #18446 made a mess with ForeignIndex and triggered a design flaw/bug of #16356, then a quick patch #21271 helped #18446, then the the bug was re-triggered by #21721 .

Related:
* #16356
    * BasicIssueContext https://github.com/go-gitea/gitea/pull/16356/files#diff-7938eb670d42a5ead6b08121e16aa4537a4d716c1cf37923c70470020fb9d036R16-R27
* #18446 
    * If some issues were dumped without ForeignIndex, then they would be imported as ForeignIndex=0  https://github.com/go-gitea/gitea/pull/18446/files#diff-1624a3e715d8fc70edf2db1630642b7d6517f8c359cc69d58c3958b34ba4ce5eR38-R39
* #21271
    * It patched the above bug (somewhat), made the issues without ForeignIndex could have the same value as LocalIndex
* #21721 
    * It re-triggered the zero-ForeignIndex bug.

ps: I am not sure whether the changes in `GetForeignIndex` are ideal (at least, now it has almost the same behavior as BasicIssueContext in #16356), it's just a quick fix. Feel free to edit on this PR directly or replace it.

### CODE REVIEW

**Review by lunny — APPROVED** [review]

**Review by zeripath — APPROVED** [review]

**Comment by lunny** [comment]
LG-TM bot
